optimum/onnxruntime/modeling_ort.py

```diff
@@ -0,0 +1,681 @@
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~OnnxModel.from_pretrained`] method to load the model weights.
+        model ([`onnxruntime.InferenceSession`](https://onnxruntime.ai/docs/api/python/api_summary.html#inferencesession)): This is the main class used to run a model. Check out the [`~OnnxModel.load_model`]
+        for more information.
+"""
+
+ONNX_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (`torch.Tensor` of shape `({0})`):
+            Indices of input sequence tokens in the vocabulary.
+            Indices can be obtained using [`AutoTokenizer`](https://huggingface.co/docs/transformers/autoclass_tutorial#autotokenizer). 
+            See [`PreTrainedTokenizer.encode`](https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizerBase.encode) and
+            [`PreTrainedTokenizer.__call__`](https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizerBase.__call__) for details.
+            [What are input IDs?](https://huggingface.co/docs/transformers/glossary#input-ids)
+        attention_mask (`torch.Tensor` of shape `({0})`, *optional*):
+            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+            - 1 for tokens that are **not masked**,
+            - 0 for tokens that are **masked**.
+            [What are attention masks?](https://huggingface.co/docs/transformers/glossary#attention-mask)
+        token_type_ids (`torch.Tensor` of shape `({0})`, *optional*):
+            Segment token indices to indicate first and second portions of the inputs. Indices are selected in `[0, 1]`:
+            - 1 for tokens that are **sentence A**,
+            - 0 for tokens that are **sentence B**.
+            [What are token type IDs?](https://huggingface.co/docs/transformers/glossary#token-type-ids)
+"""
+
+
+@add_start_docstrings(
+    """
+    Base OnnxModel class for implementing models using ONNX Runtime. The OnnxModel implements generic methods for interacting
+    with the Hugging Face Hub as well as exporting vanilla transformers models to ONNX using `transformers.onnx` toolchain.
+    The OnnxModel implements additionally generic methods for optimizing and quantizing Onnx models.
+    """,
+)
+class OnnxModel(OptimizedModel):
+    base_model_prefix = "onnx_model"
+
+    def __init__(self, model=None, config=None, **kwargs):
+        self.model = model
+        self.config = config
+        self.model_save_dir = kwargs.get("model_save_dir", None)
+        self.latest_model_name = kwargs.get("latest_model_name", "model.onnx")
+
+    def forward(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def optimize(
```

Comment:
A few questions here:

- How does this collide with what is being done in terms of optmization in `optimu.onnxruntime`?
- Personally, I'm not sure we should embed features such as export, optimization and quantization in the `OnnxModel` class. I see more this class as a direct wrapper of all the ORT machinery for inference: it takes an ONNX model and performs the forward pass while making sure to respect the transformers API.

@echarlaix @JingyaHuang @mfuntowicz @regisss wdyt?

Reply:
I agree with @michaelbenayoun here that it will be tricky to maintain a universal `optimize()` and `quantize()` function, especially for techniques like static quantization where some configuration is needed to make it work. 

I think this approach would also depart from the "fit-export" API that's already been developed in `optimum`, so it might be confusing for users on which API they should use

Reply:
Happy to discuss this tomorrow and then comment on the result of it here. I implemented these methods as a quick way of showing how they could work. My idea was to reuse `ORTOptimizer` inside `optimize` to still have the `model.optimize` but only maintain the `ORTOptimizer`.

Reply:
decided to remove for now